eureka2-read-server/src/main/java/com/netflix/eureka2/server/registry/EurekaReadServerRegistry.java

```diff
@@ -80,9 +83,19 @@ public int size() {
         throw new UnsupportedOperationException("method not supported by EurekaReadServerRegistry");
     }
 
+    // TODO As read server is based on client API, and is doing fullRegistryFetch from write server, we cannot generate batch markers4
+    // TODO This is one of the reasons why we cannot depend on Eureka client API in the read server.
     @Override
-    public Observable<ChangeNotification<InstanceInfo>> forInterest(Interest<InstanceInfo> interest) {
-        return eurekaClient.forInterest(interest);
+    public Observable<ChangeNotification<InstanceInfo>> forInterest(final Interest<InstanceInfo> interest) {
+        return eurekaClient.forInterest(interest).map(new Func1<ChangeNotification<InstanceInfo>, ChangeNotification<InstanceInfo>>() {
+            @Override
+            public ChangeNotification<InstanceInfo> call(ChangeNotification<InstanceInfo> notification) {
```

Comment:
Instead of map.filterNull can we not just use the special filter already created for this purposes?